packages/cubejs-schema-compiler/scaffolding/ScaffoldingSchema.js

```diff
@@ -148,10 +148,15 @@ class ScaffoldingSchema {
   };
 
   joins(tableName, tableDefinition) {
+    const key = "id";
     return R.unnest(tableDefinition
-      .filter(column => column.name.toLowerCase().indexOf('id') !== -1 && column.name.toLowerCase() !== 'id')
+      .filter(column => {
+        const columnName = column.name.toLowerCase();
+        const columnKeyPosition = columnName.length - columnName.lastIndexOf(key) - key.length;
+        return (columnKeyPosition === 0 && columnName !== 'id')
```

Comment:
This is right but missing `_id` case I guess.

Reply:
Probably just makes sense to use `RegExp` everywhere.

Reply:
Correct, RegExp similar to done to the below would simplify.

Reply:
Updated,  please check if it works.